core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -74,7 +74,17 @@ object StaticFile {
       }
     })
 
-  def calcETag[F[_]: Sync]: File => F[String] = f => Sync[F].delay(f.length().toString)
+  def calcETag[F[_]: Sync]: File => F[String] =
+    f =>
+      Sync[F]
+        .delay(f.isFile)
+        .flatMap { isFile =>
+          if (isFile) {
+            Sync[F].delay(s"${f.lastModified()}${f.length()}")
+          } else {
+            Sync[F].pure("")
```

Comment:
Is this what we want, or do we want to omit the header?  I don't think an empty `Etag` makes any sense.  Maybe the calculator should return an `Option`.

Reply:
It doesn't matter, if it's not a file the whole this is never called anyway. Nothing is added.